lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -168,6 +168,200 @@ class ConvertAtenEmbeddingOp : public OpConversionPattern<AtenEmbeddingOp> {
+                                          //get offsets[i+1]                                  
+                                          Value index_i_1    = castIntToIndex(rewriter, loc, index_i_1_int);
+                                          Value checkLast    = rewriter.create<arith::CmpIOp>(loc, arith::CmpIPredicate::eq, castIndexToInt64(rewriter, loc, offsetsLength), index_i_1_int); 
+                                          Value offsets_i_1  = rewriter.create<arith::SelectOp>(loc, checkLast, castIndexToInt64(rewriter, loc, indicesLength),  rewriter.create<tensor::ExtractOp>(loc, offsets, index_i_1)); 
+
+                                          //get j
+                                          Value index_j = castIndexToInt64(rewriter, loc,  rewriter.create<linalg::IndexOp>(loc, /*value*/1));
+
+                                          //offsets[i] <=j
+                                          Value lessThan0 = rewriter.create<arith::CmpIOp>(loc, arith::CmpIPredicate::slt, offsets_i, index_j);
+                                          Value eq0       = rewriter.create<arith::CmpIOp>(loc, arith::CmpIPredicate::eq, offsets_i, index_j); 
+                                          Value and0       = rewriter.create<arith::OrIOp>(loc, lessThan0, eq0);
+
+                                          //j < offsets[i+1]
+                                          Value lessThan   = rewriter.create<arith::CmpIOp>(loc, arith::CmpIPredicate::slt, index_j, offsets_i_1);
+
+                                          //offsets[i] <= j and j < offsets[i+1]
+                                          Value netResult = rewriter.create<arith::AndIOp>(loc, and0, lessThan);
+
+                                          //get the indices into the weight tensor
+                                          SmallVector<Value> index_into_weight;
+                                          index_into_weight.push_back(castIntToIndex(rewriter, loc, index_in_indices));
+                                          index_into_weight.push_back(rewriter.create<linalg::IndexOp>(loc, /*value=*/2));                                  
+                                          Value weightElem = rewriter.create<tensor::ExtractOp>(loc, weight, index_into_weight);
+
+                                          Value addResult = rewriter.create<arith::AddFOp>(loc, weightElem, initTensorElem);
+                                          Value select = rewriter.create<arith::SelectOp>(loc, netResult, addResult, initTensorElem);
+                                          rewriter.create<linalg::YieldOp>(loc, select);
+
+                                        }).getResult(0);
+
+      //cast outputType. 
+      auto resultType1 = typeConverter->convertType(op->getResult(0).getType());
+      Value initTensor_new = rewriter.create<tensor::CastOp>(loc, resultType1, embeddingBagResult);
+
+      //offset2 tensor, this should be an empty tensor for the sum mode
```

Comment:
is the tensor supposed to be empty or zero?

Reply:
It should be an empty tensor.